cirq/linalg/decompositions.py

```diff
@@ -854,3 +857,98 @@ def kak_decomposition(unitary_object: Union[np.ndarray, 'cirq.SupportsUnitary'],
         global_phase=g * inner_cannon.global_phase,
         single_qubit_operations_before=(b1, b0),
         single_qubit_operations_after=(a1, a0))
+
+
+def kak_vector(unitary: np.ndarray):
+    r"""Compute the KAK vectors of a two qubit unitary.
+
+    Any 2 qubit unitary may be expressed as
+
+    $$ U = k_l A k_r $$
+    where $k_l, k_r$ are single qubit (local) unitaries and
+    $$ A = \exp( i * \sum_{j=x,y,z} k_j \sigma_{j,0}\sigma{j,1} ) $$
+
+    where $(k_x,k_y,k_z)$ is in the Weyl chamber, a tetrahedron with corners
+    $$(0,0,0), (\pi/2,0,0), (\pi/4,pi/4,0), (\pi/4,\pi/4,\pi/4)$$
+
+    The vector entries are ordered such that
+    $$ \pi/2 - k_y \geq k_x \geq k_y \geq z \geq 0 $$
```

Comment:
The canonicalization we use is actually 

```
0 ≤ abs(z2) ≤ y2 ≤ x2 ≤ π/4
if x2 = π/4, z2 >= 0
```

This canonicalization has the benefit that operations that are near the identity are near (0, 0, 0) instead of possibly being near (pi/2, 0, 0). And the only place where there's a discontinuity vs the 3d euclidean distance is on the x=max face, where z can flip sign from small perturbations to the unitary matrix.

Reply:
Changed the canonicalization to match the one above.